sources/engine/Stride.Graphics/Direct3D/SwapChainGraphicsPresenter.Direct3D.cs

```diff
@@ -56,13 +60,39 @@ public SwapChainGraphicsPresenter(GraphicsDevice device, PresentationParameters
         {
             PresentInterval = presentationParameters.PresentationInterval;
 
+            flipModelSupport = CheckFlipModelSupport(device);
+            tearingSupport = CheckTearingSupport(device);
+
             // Initialize the swap chain
             swapChain = CreateSwapChain();
 
             backBuffer = new Texture(device).InitializeFromImpl(swapChain.GetBackBuffer<BackBufferResourceType>(0), Description.BackBufferFormat.IsSRgb());
 
             // Reload should get backbuffer from swapchain as well
             //backBufferTexture.Reload = graphicsResource => ((Texture)graphicsResource).Recreate(swapChain.GetBackBuffer<SharpDX.Direct3D11.Texture>(0));
+
+            static bool CheckFlipModelSupport(GraphicsDevice device)
+            {
+                // From https://github.com/walbourn/directx-vs-templates/blob/main/d3d11game_win32_dr/DeviceResources.cpp#L138
+                using var dxgiDevice = device.NativeDevice.QueryInterface<SharpDX.DXGI.Device>();
+                using var dxgiAdapter = dxgiDevice.Adapter;
+                using var dxgiFactory = dxgiAdapter.GetParent<SharpDX.DXGI.Factory4>();
+                return dxgiFactory != null;
+            }
+
+            static unsafe bool CheckTearingSupport(GraphicsDevice device)
+            {
+                // From https://learn.microsoft.com/en-us/windows/win32/direct3ddxgi/variable-refresh-rate-displays
+                using var dxgiDevice = device.NativeDevice.QueryInterface<SharpDX.DXGI.Device>();
+                using var dxgiAdapter = dxgiDevice.Adapter;
+                using var dxgiFactory = dxgiAdapter.GetParent<SharpDX.DXGI.Factory5>();
```

Comment:
I suspect this would also need to be wrapped in try-catch.

Reply:
Thanks, try/catch has been added.